src/deadline/cinema4d_adaptor/Cinema4DAdaptor/adaptor.py

```diff
@@ -298,12 +303,70 @@ def _start_cinema4d_client(self) -> None:
             new_module_path = plugin_dir + os.pathsep + module_path
         os.environ[module_path_key] = new_module_path
 
+        arguments = [c4d_exe, "-nogui", "-DeadlineCloudClient"]
+        if "linux" in platform.system().lower():
+            _logger.info("Setting Linux Cinema4D environment")
+            env = self._get_cinema4d_environment(c4d_exe)
+            _logger.info("Inserting Linux adaptor wrapper script")
+            arguments.insert(0, os.path.join(os.path.dirname(__file__), "adaptor.sh"))
+
+        if os.environ.get("g_licenseServerRLM", None):
+            arguments.append("g_licenseServerRLM={}".format(os.environ.get("g_licenseServerRLM")))
+        if os.environ.get("g_licenseServerUrl", None):
+            arguments.append("g_licenseServerUrl={}".format(os.environ.get("g_licenseServerUrl")))
+
         self._cinema4d_client = LoggingSubprocess(
-            args=[c4d_exe, "-nogui", "-DeadlineCloudClient"],
+            args=arguments,
             stdout_handler=regexhandler,
             stderr_handler=regexhandler,
         )
 
+    @staticmethod
+    def _glob_add_path(base, pattern):
+        matches = list(glob.iglob(os.path.join(base, pattern)))
+        if matches:
+            return os.path.join(base, matches[0])
+        return base
+
+    def _get_cinema4d_environment(self, c4d_exe):
```

Comment:
Similar comment to the one I have above about the differences between the this and the `adaptor.sh`.

If we do want to keep this function, we should add `linux` to the function name, and document as a docstring what this tackles that the wrapper script doesn't.

Reply:
Good catch. I'll move it over to the wrapper